newIDE/app/src/InstancesEditor/index.js

```diff
@@ -777,25 +802,43 @@ export default class InstancesEditor extends Component<Props> {
     sceneY: number
   ) => {
     this.fpsLimiter.notifyInteractionHappened();
+
+    this.hasCursorMovedSinceItIsDown = false;
+
     if (this.keyboardShortcuts.shouldMoveView()) {
       // If the user wants to move the view, discard the click on an instance:
       // it's just the beginning of the user panning the view.
       return;
     }
 
+    if (
+      this.keyboardShortcuts.shouldStartRectangleSelectionInsteadOfSelecting()
+    ) {
+      const canvasPosition = this.viewPosition.toCanvasCoordinates(
+        sceneX,
+        sceneY
+      );
+      this.selectionRectangle.startSelectionRectangle(
+        canvasPosition[0],
+        canvasPosition[1]
+      );
+      return;
+    }
+
+    // MultiSelect is not done here because it's the same modifier as
```

Comment:
What do you mean by "MultiSelect is not done here" ? Because I see just after `multiSelect: this.keyboardShortcuts.shouldMultiSelect(),` so I'm confused.

Reply:
`shouldMultiSelect()` and `shouldStartRectangleSelectionInsteadOfSelecting()` are both `true` when Shift is held so at this point `shouldMultiSelect()` can only be `false`.

Reply:
(so I was wondering if this should be false)

Reply:
I though that a comment was better because if we move these lines for some reason, we could miss that we may not want to always set it to false and make a regression.